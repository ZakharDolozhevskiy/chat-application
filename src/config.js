export const api = {
  host: 'http://185.13.90.140:8081/',
  channel: 'message'
};

export const routes = {
  MAIN: '/',
  SETTINGS: '/settings'
};