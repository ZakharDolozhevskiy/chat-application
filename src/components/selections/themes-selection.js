import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Radio from '@material-ui/core/es/Radio/Radio';
import Typography from '@material-ui/core/Typography';
import RadioGroup from '@material-ui/core/es/RadioGroup/RadioGroup';
import FormControlLabel from '@material-ui/core/es/FormControlLabel/FormControlLabel';

import { baseStyles } from './styles';
import { themes } from '../../config';

class ThemesSelection extends React.PureComponent {
  onChange = (event, theme) =>
    this.props.onChange(theme);

  render() {
    const { className, translate, current } = this.props;

    return (
      <div className={className}>
        <Typography component="p" color="inherit" className="label">
          {translate('Interface colors')}
        </Typography>
        <RadioGroup value={current} onChange={this.onChange} className="controls">
          <FormControlLabel
            value={themes.LIGHT}
            label={translate('Light')}
            control={<Radio className="selection-radio"/>}
          />
          <FormControlLabel
            value={themes.DARK}
            label={translate('Dark')}
            control={<Radio className="selection-radio"/>}
          />
        </RadioGroup>
      </div>
    );
  }
}

ThemesSelection.propTypes = {
  current: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  className: PropTypes.string
};

export default styled(ThemesSelection)`${baseStyles}`;