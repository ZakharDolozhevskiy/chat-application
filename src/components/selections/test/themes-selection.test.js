import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { ThemesSelection } from '../themes-selection';

import { themes } from '../../../config';

describe('<ShortcutsSelection/>', () => {
  const callback = jest.fn();

  const component = shallow(
    <ThemesSelection
      className=""
      onChange={callback}
      translate={id => id}
      currentTheme={themes.LIGHT}
    />
  );

  it('should render component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call onChange callback if changes happen', () => {
    component.instance().onChange(null, themes.DARK);
    expect(callback.mock.calls[0][0]).toBe(themes.DARK);
  });
});