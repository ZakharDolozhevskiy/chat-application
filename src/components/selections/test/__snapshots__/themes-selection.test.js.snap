// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShortcutsSelection/> should render component 1`] = `
<div
  className=""
>
  <WithStyles(Typography)
    className="label"
    color="inherit"
    component="p"
  >
    Interface colors
  </WithStyles(Typography)>
  <RadioGroup
    className="controls"
    onChange={[Function]}
    value="light"
  >
    <WithStyles(FormControlLabel)
      control={
        <WithStyles(Radio)
          className="selection-radio"
        />
      }
      label="Light"
      value="light"
    />
    <WithStyles(FormControlLabel)
      control={
        <WithStyles(Radio)
          className="selection-radio"
        />
      }
      label="Dark"
      value="dark"
    />
  </RadioGroup>
</div>
`;
